client/blocks/taxonomy-manager/list-item.jsx

```diff
@@ -56,8 +58,11 @@ class TaxonomyManagerListItem extends Component {
 
 	closeDeleteDialog = action => {
 		if ( action === 'delete' ) {
-			const { siteId, taxonomy, term } = this.props;
-			this.props.deleteTerm( siteId, taxonomy, term.ID, term.slug );
+			const { siteId, taxonomy, term, afterDelete } = this.props;
+			this.props.deleteTerm( siteId, taxonomy, term.ID, term.slug )
+				.then( () => {
+					afterDelete( term );
+				} );
```

Comment:
I'm a pragmatic person so I personally don't mind using the returned promise from Redux action creators when it seems to be the simplest solution like here, but I know this breaks the unidirectional redux data flow (which @aduth disagrees with). I think a mix between a flag `itemToBeDeleted` in the component state and using `componentWillUnmount` to trigger `afterDelete` could have the same behavior.

Thoughts on this @aduth ?